### PR TITLE
Incremental obfuscation functionality

### PR DESCRIPTION
**Pull request by aasakawa**
Since ProGuard supports [incremental obfuscation](https://www.guardsquare.com/proguard/manual/examples#incremental) by specifying `-applymapping`, I propose adding a functionality to incrementally obfuscate projects with multiple modules.

- `root:pom`
  - `test1:jar`
    - `test.Class1`
      - `void method(test.Class1)`
    - `test.Keep1`
  - `test2:jar`
    - `test.Class2`
      - `void method(test.Class1)`
    - `test.Keep2`

(An example of projects that actually work: [incremental-obfuscation.zip](https://github.com/wvengen/proguard-maven-plugin/files/2025747/incremental-obfuscation.zip))

In the project of the above layout, If you specify `proguard_map.txt` which is the result of obfuscating `test1:jar` with `-applymapping` when obfuscating `test2:jar`, ProGuard will consider the result of changing the identifier in `test1:jar`, so `test.Class1` and `test.Class2` will not change to the same name.

In order to support such incremental obfuscation, I introduce the following two parameters.

- `File applyMappingFile (default: none)`
  - If specified, specify that file as `-applymapping` parameter
- `boolean incremental (default: false)`
  - Specify whether to perform incremental obfuscation
  - In the case of `true`, the following behavior changes
    1. Even if `applyMappingFile` is specified, if the file does not exist, omit `-applymapping` (i.e. there is no mapping file to be applied to the originating project)
    2. If the file exists, merge the mapping file of the execution result of ProGuard into the file specified by `applyMappingFile`

In other words, if you set `incremental = true`, you will be able to obfuscate individual modules one by one in projects with multiple modules.

Note that this functionality does not work properly in versions prior to ProGuard 5.3.

>#### Version 5.3 (Sep. 2016)
>- Avoiding obfuscated name clashes with library classes.

### CODE REVIEW

**Comment by lasselindqvist** [comment]
@aasakawa seems good to me, but as this a somewhat complex functionality, it would benefit from good documentation, maybe an example. The explanation in this pull request is good, could it be added in some form to the site (index.apt)?

**Comment by MarcMil** [comment]
@lasselindqvist Here is a link to an example https://github.com/wvengen/proguard-maven-plugin/files/2025747/incremental-obfuscation.zip as it has been posted by @aasakawa

**Comment by lasselindqvist** [comment]
I will add the documentation later in #85
